### PR TITLE
Upgrade Storybook (v6)

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons/issues/13813

Replaces https://github.com/mozilla/addons-frontend/pull/9603

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -298,10 +298,9 @@
     "@emotion/core": "^10.0.16",
     "@loadable/babel-plugin": "^5.7.1",
     "@loadable/webpack-plugin": "^5.2.1",
-    "@storybook/addon-info": "5.3.21",
```
Comment (willdurand) [inline package.json:301]: addon-info is deprecated for quite some time.. We were using it for displaying props of some components but it didn't always work well (and had issues with Flow in the past). The new addon-docs seems better.

File: src/ui/components/PromotedBadge/index.js [hunk 2]
```diff
@@ -80,7 +80,7 @@ export const PromotedBadgeBase = ({
         target="_blank"
         title={linkTitle}
       >
-        <IconPromotedBadge category={category} size={size} />
+        <IconPromotedBadge category={category} i18n={i18n} size={size} />
```
Comment (willdurand) [inline src/ui/components/PromotedBadge/index.js:83]: this story is currently broken because of this

File: stories/amo/HeroRecommendation.js [hunk 3]
```diff
@@ -40,6 +40,7 @@ const render = (
 };
 
 storiesOf('HeroRecommendation', module)
+  .addParameters({ component: HeroRecommendationBase })
```
Comment (willdurand) [inline stories/amo/HeroRecommendation.js:43]: needed to get props descriptions

File: stories/setup/main.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        rules: [
+          ...config.module.rules,
+          ...getStyleRules({ bundleStylesWithJs: true }),
+        ],
+      },
+      plugins: [
+        ...config.plugins,
+        ...getPlugins({ includeLoadablePlugin: false }),
+      ],
+      resolve: webpackProdConfig.resolve,
```
Comment (willdurand) [inline stories/setup/main.js:27]: this is more or less what we had in the webpack config files before but now storybook wants us to declare the config here.

**Comment by willdurand** [comment]
(rebased)

**Review by bobsilverberg — APPROVED** [review]
Looks good, thanks for taking this on @willdurand!

**Pull request merged**